src/vsr/grid.zig

```diff
@@ -549,13 +584,13 @@ pub fn GridType(comptime Storage: type) type {
             const block_header = schema.header_from_block(block.*);
             assert(grid.superblock.opened);
             assert(grid.canceling == null);
-            assert(grid.checkpointing == null);
+            maybe(grid.checkpointing != null); // FreeSetEncoded writes a block.
             assert(grid.writing(block_header.address, block.*) == .not_writing);
             assert(!grid.blocks_missing.repair_waiting(
                 block_header.address,
                 block_header.checksum,
             ));
-            assert(!grid.superblock.free_set.is_free(block_header.address));
```

Comment:
This assert is what motivated lifting `FreeSet.checkpoint()` call from `FreeSetEncoded` to the grid itself.